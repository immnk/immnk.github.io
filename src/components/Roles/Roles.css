.root {
  margin: 0 auto;
  margin-top: -12rem;
  padding: 0.8rem 2.4rem 2.4rem;
}

.container {
  background: #fffff9;
  color: #676767;
  display: flex;
  font-size: 1.4rem;
  border-top-left-radius: 40px;
  border-top-right-radius: 40px;
  max-width: 1200px;
  margin: 0 auto;
  box-shadow: 0 5px 5px 0 rgb(233 240 243 / 50%), 0 0 0 1px #e6ecf8;
}

.subContainer {
  flex: 1;
  text-align: center;
  border-bottom: 1px solid #e6ecf8;
  border-right: 1px solid #e6ecf8;
  padding: 6rem 3rem;

  &:last-child {
    border-bottom: none;
    border-right: none;
  }
}

.mobile {
}

.frontend {
}

.node {
}

.icon {
  font-size: 5rem;
  color: #bfacaa;
}

.description {
  padding: 2rem 1rem;

  code {
    font-family: monospace;
  }
}

.title {
  font-size: 2.4rem;
  margin-top: 1.5rem;
}

.listTitle {
  color: #bfacaa;
  font-size: 1.5rem;
}

.list {
  list-style-type: none;
  padding: 1.5rem;
}
