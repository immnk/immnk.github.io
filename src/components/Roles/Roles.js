import React from 'react';
import PropTypes from 'proptypes';
import classNames from 'classnames';
import ScrollAnimation from 'react-animate-on-scroll';

// eslint-disable-next-line css-modules/no-unused-class
import css from './Roles.css';

// eslint-disable-next-line max-lines-per-function
const Roles = ({ className }) => {
  const myRoles = [
    {
      id: 'mobile',
      icon: 'fas fa-mobile-alt',
      title: 'Mobile Developer',
      description: `I have developed enterprise hybrid mobile applications for <code>Android</code> and
  <code>iOS</code>. I have also developed some apps for personal use. Do check out
  <a
    href="https://play.google.com/store/apps/details?id=com.creatuslabs.portfolio"
    rel="noreferrer"
  >
    here
  </a>`,
      tools: ['Ionic', 'Cordova', 'Swift', 'Java'],
      domains: ['eCommerce', 'Insurance', 'Gaming']
    },
    {
      id: 'frontend',
      icon: 'fas fa-laptop-code',
      title: 'Front-end Developer',
      description: `I like to build user interfaces for applications that solve problems for real users 
        and turn ideas into reality!`,
      tools: [
        'ReactJS',
        'Vanilla JS',
        'HTML5',
        'CSS3',
        'GitHub',
        'iTerm',
        'VS Code',
        'CSS Modules'
      ],
      domains: ['Banking', 'Insurance', 'Trading']
    },
    {
      id: 'node',
      icon: 'fab fa-node-js',
      title: 'NodeJS Developer',
      description:
        'I write backend code using NodeJS to create a full stack apps that can communicate with mobile or web.',
      tools: ['NodeJS', 'ExpressJS', 'mongoDB', 'MERN'],
      domains: ['eCommerce', 'Insurance']
    }
  ];
  return (
    <section className={classNames(css.root, className)}>
      <div className={css.container}>
        {myRoles.map((role) => {
          return (
            <ScrollAnimation
              animateIn="pulse"
              key={role.id}
              className={classNames(css.subContainer, css[role.id])}
            >
              <figure className={css.icon}>
                <i className={role.icon} />
              </figure>
              <h1 className={css.title}>{role.title}</h1>
              <p
                className={css.description}
                // eslint-disable-next-line react/no-danger
                dangerouslySetInnerHTML={{ __html: role.description }}
              />
              <p className={css.listTitle}>Tools used</p>
              <ul className={css.list}>
                {role.tools.map((tool) => (
                  <li key={tool}>{tool}</li>
                ))}
              </ul>
              <p className={css.listTitle}>Domains Worked</p>
              <ul className={css.list}>
                {role.domains.map((domain) => (
                  <li key={domain}>{domain}</li>
                ))}
              </ul>
            </ScrollAnimation>
          );
        })}
      </div>
    </section>
  );
};

Roles.defaultProps = {
  className: undefined
};

Roles.propTypes = {
  className: PropTypes.string
};

export default Roles;
