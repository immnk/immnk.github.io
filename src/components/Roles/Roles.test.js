import React from 'react';
import { shallow } from 'enzyme';

import Roles from './Roles';

const renderApp = (props) => {
  const defaultProps = {
    className: 'my-classname'
  };

  return <Roles {...defaultProps} {...props} />;
};

const shallowRenderApp = (props) => shallow(renderApp(props));

describe('Test App', () => {
  it('render app', () => {
    const actual = shallowRenderApp();
    expect(actual).toMatchSnapshot();
  });
});
