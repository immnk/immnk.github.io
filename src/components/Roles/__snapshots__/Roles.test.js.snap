// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App render app 1`] = `
<section
  className="root my-classname"
>
  <div
    className="container"
  >
    <ScrollAnimation
      animateIn="pulse"
      animateOnce={false}
      animatePreScroll={true}
      className="subContainer mobile"
      delay={0}
      duration={1}
      initiallyVisible={false}
      key="mobile"
      offset={150}
    >
      <figure
        className="icon"
      >
        <i
          className="fas fa-mobile-alt"
        />
      </figure>
      <h1
        className="title"
      >
        Mobile Developer
      </h1>
      <p
        className="description"
        dangerouslySetInnerHTML={
          Object {
            "__html": "I have developed enterprise hybrid mobile applications for <code>Android</code> and
  <code>iOS</code>. I have also developed some apps for personal use. Do check out
  <a
    href=\\"https://play.google.com/store/apps/details?id=com.creatuslabs.portfolio\\"
    rel=\\"noreferrer\\"
  >
    here
  </a>",
          }
        }
      />
      <p
        className="listTitle"
      >
        Tools used
      </p>
      <ul
        className="list"
      >
        <li
          key="Ionic"
        >
          Ionic
        </li>
        <li
          key="Cordova"
        >
          Cordova
        </li>
        <li
          key="Swift"
        >
          Swift
        </li>
        <li
          key="Java"
        >
          Java
        </li>
      </ul>
      <p
        className="listTitle"
      >
        Domains Worked
      </p>
      <ul
        className="list"
      >
        <li
          key="eCommerce"
        >
          eCommerce
        </li>
        <li
          key="Insurance"
        >
          Insurance
        </li>
        <li
          key="Gaming"
        >
          Gaming
        </li>
      </ul>
    </ScrollAnimation>
    <ScrollAnimation
      animateIn="pulse"
      animateOnce={false}
      animatePreScroll={true}
      className="subContainer frontend"
      delay={0}
      duration={1}
      initiallyVisible={false}
      key="frontend"
      offset={150}
    >
      <figure
        className="icon"
      >
        <i
          className="fas fa-laptop-code"
        />
      </figure>
      <h1
        className="title"
      >
        Front-end Developer
      </h1>
      <p
        className="description"
        dangerouslySetInnerHTML={
          Object {
            "__html": "I like to build user interfaces for applications that solve problems for real users 
        and turn ideas into reality!",
          }
        }
      />
      <p
        className="listTitle"
      >
        Tools used
      </p>
      <ul
        className="list"
      >
        <li
          key="ReactJS"
        >
          ReactJS
        </li>
        <li
          key="Vanilla JS"
        >
          Vanilla JS
        </li>
        <li
          key="HTML5"
        >
          HTML5
        </li>
        <li
          key="CSS3"
        >
          CSS3
        </li>
        <li
          key="GitHub"
        >
          GitHub
        </li>
        <li
          key="iTerm"
        >
          iTerm
        </li>
        <li
          key="VS Code"
        >
          VS Code
        </li>
        <li
          key="CSS Modules"
        >
          CSS Modules
        </li>
      </ul>
      <p
        className="listTitle"
      >
        Domains Worked
      </p>
      <ul
        className="list"
      >
        <li
          key="Banking"
        >
          Banking
        </li>
        <li
          key="Insurance"
        >
          Insurance
        </li>
        <li
          key="Trading"
        >
          Trading
        </li>
      </ul>
    </ScrollAnimation>
    <ScrollAnimation
      animateIn="pulse"
      animateOnce={false}
      animatePreScroll={true}
      className="subContainer node"
      delay={0}
      duration={1}
      initiallyVisible={false}
      key="node"
      offset={150}
    >
      <figure
        className="icon"
      >
        <i
          className="fab fa-node-js"
        />
      </figure>
      <h1
        className="title"
      >
        NodeJS Developer
      </h1>
      <p
        className="description"
        dangerouslySetInnerHTML={
          Object {
            "__html": "I write backend code using NodeJS to create a full stack apps that can communicate with mobile or web.",
          }
        }
      />
      <p
        className="listTitle"
      >
        Tools used
      </p>
      <ul
        className="list"
      >
        <li
          key="NodeJS"
        >
          NodeJS
        </li>
        <li
          key="ExpressJS"
        >
          ExpressJS
        </li>
        <li
          key="mongoDB"
        >
          mongoDB
        </li>
        <li
          key="MERN"
        >
          MERN
        </li>
      </ul>
      <p
        className="listTitle"
      >
        Domains Worked
      </p>
      <ul
        className="list"
      >
        <li
          key="eCommerce"
        >
          eCommerce
        </li>
        <li
          key="Insurance"
        >
          Insurance
        </li>
      </ul>
    </ScrollAnimation>
  </div>
</section>
`;
